File: MergeTwoLists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeTwoLists(ListNode list1, ListNode list2) {
        
        // Default
        if (list1 == null && list2 == null)
        {
            return null;
        }
        
        List<int> Nodes = new List<int>();
        ListNode Output = new ListNode();
        
        while ((list1 != null) && (list2 != null))
        {
            if (list1.val <= list2.val)
            {
                Nodes.Add(list1.val);
                list1 = list1.next;
            }
            else
            {
                Nodes.Add(list2.val);
                list2 = list2.next;
            }
        }
        
        while (list1 != null)
        {
            Nodes.Add(list1.val);
            list1 = list1.next;
        }
        
        while (list2 != null)
        {
            Nodes.Add(list2.val);
            list2 = list2.next;
        }
        
        ListNode butt = new ListNode(Nodes[Nodes.Count - 1], null);
        
        for (int i = Nodes.Count - 2; i > -1; i--)
        {
            Output = new ListNode(Nodes[i], butt);
            butt = Output;
        }
        
        return Output;
        
    }
}

File: Python/mergeTwoLists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:

        valList = []

        while list1 and list2:

            if (list1.val < list2.val):
                valList.append(list1.val)
                list1 = list1.next
            else:
                valList.append(list2.val)
                list2 = list2.next

        while list1:
            valList.append(list1.val)
            list1 = list1.next

        while list2:  
            valList.append(list2.val)
            list2 = list2.next

        valList.reverse()

        curr, prev = None, None

        for val in valList:
            curr = ListNode(val, prev)
            prev = curr

        return curr
